sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -1175,6 +1205,27 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     client.listFunctions(db, pattern)
   }
 
+  /** Detect whether a table is stored with Hive-compatible metadata. */
+  private def isHiveCompatible(table: CatalogTable): Boolean = {
```

Comment:
And, also, if you look at `createDataSourceTable`, it does more checks that your example to decide whether a table is Hive-compatible or not. And the ultimate check is whether the Hive metastore accepts the table (exception means Spark tries to create the table again, as non-compatible).

So, again, you cannot detect compatibility by just checking the schema.